avocado/core/messages.py

```diff
@@ -157,6 +159,51 @@ def handle(self, message, task, job):
 class BaseRunningMessageHandler(BaseMessageHandler):
     """Base interface for resolving running messages."""
 
+    _tag = b''
+
+    def __init__(self):
+        self.line_buffer = b''
+
+    def _get_tagged_line(self, data, encoding=None):
+        """
+        It will add the tag in to each line of data as a prefix. If the data
+        don't finish with the new line character, the last line is marked
+        as incomplete, and it is saved to the line buffer for next usage.
```

Comment:
This last part that I'm talking about on a following comment. Maybe we could let each method do only one single thing and the split/check for incomplete lines could be done in a separated method.

Reply:
Ok, I will refactor this.